src/Databases/DatabasesCommon.cpp

```diff
@@ -166,4 +168,66 @@ StoragePtr DatabaseWithOwnTablesBase::getTableUnlocked(const String & table_name
                     backQuote(database_name), backQuote(table_name));
 }
 
+/// Daisy: starts.
+StorageInMemoryCreateQuery parseCreateQueryFromAST(const ASTPtr & query, const String & database_, const String & table_)
+{
+    return parseCreateQueryFromAST(query.get(), database_, table_);
+}
+
+StorageInMemoryCreateQuery parseCreateQueryFromAST(const IAST * query, const String & database_, const String & table_)
```

Comment:
do we need a `const IAST * query` version  function of `parseCreateQueryFromAST` ? Why not just have the version of `const ASTPtr & query` ?

Reply:
I added the `const IAST * query`  version function for the following code
```
    void tryAttachTable(
        ContextPtr context,
        const ASTCreateQuery & query,
        DatabaseOrdinary & database,
        const String & database_name,
        const String & metadata_path,
        bool has_force_restore_data_flag)
    {
        try
        {
            auto [table_name, table] = createTableFromAST(
                query,
                database_name,
                database.getTableDataPath(query),
                context,
                 has_force_restore_data_flag);

             database.attachTable(table_name, table, database.getTableDataPath(query));

             /// Daisy: starts.
             const auto & new_create_query = parseCreateQueryFromAST(&query, database_name, table_name);
             table->setInMemoryCreateQuery(new_create_query);
             /// Daisy: ends.
```